tests/02-list_valid_components.bats

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bats
+
+load libs/shared_setup
+
+setup () {
+    create_git_branch MOODLE_31_STABLE v3.1.0
+
+    cd $BATS_TEST_DIRNAME/../list_valid_components/
+}
+
+@test "list_valid_components: 31 components" {
+    # We need to rewrite the directory in the fixture for verifying correct result.
+    EXPECTED=$WORKSPACE/valid_components_expected.txt
+    sed -e "s#\[gitdir\]#${gitdir}#" $BATS_TEST_DIRNAME/fixtures/31-valid_components.txt > $EXPECTED
+
+    OUTPUT=$WORKSPACE/valid_components_out.txt
+
+    # Run test
+    php list_valid_components.php --basedir=$gitdir > $OUTPUT
+    assert_success
+    diff -ruN $EXPECTED $OUTPUT
+    assert_output ''
+
+    rm $EXPECTED $OUTPUT
```

Comment:
Just guessing if:

1) This should be a call to ```clean_workspace_directory()``` instead, or, better
2) Guessing if BATS does have a corresponding teardown() method where we can automate that.

No matter of it, I'm proceeding with this, is perfect.